lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -585,16 +585,15 @@ def from_relation_data(cls, data: dict):
     def identifier(self) -> str:
         """Format the topology information into a terse string."""
         # This is odd, but may have `None` as a model key
-        return "_".join([str(val) for val in self.as_dict().values()]).replace("/", "_")
+        return "_".join([str(val) for val in self.as_promql_label_dict().values()]).replace(
+            "/", "_"
+        )
```

Comment:
Breaking change for `identifier`?

Reply:
It should be the same except without unit.

Reply:
I think we should eliminate work towards eliminating the proliferation of methods `promql_labels`, `as_promql_labels_dict` and retain only a single method `as_dict`. Hence it is may be a better idea to do something like
```
identifiers = topology.as_dict()
identifiers.pop('juju_unit')
return "_".join([str(val) for val in identifiers.values()]).replace("/", "_")
```

Reply:
@balbirthomas Please create a separate ticket for that work, unless you have already.